contk/dataloader/multi_turn_dialog.py

```diff
@@ -288,3 +291,391 @@ def _load_data(self):
+			self.key_name.append('multi_ref')
+			data['multi_ref'] = self._load_multi_ref_data()
+			data_size['multi_ref'] = len(data['multi_ref']['session'])
+		return vocab_list, len(vocab_list), data, data_size
+
+	def get_batch(self, key, index):
+		'''Get a batch of specified `index`.
+
+		Arguments:
+			key (str): must be contained in `key_name`
+			index (list): a list of specified index
+
+		Returns:
+			(dict): A dict at least contains:
+
+				* session (dict): Return of MultiTurnDialog.get_batch
+				* topics (list): A 1-d list, the topic ids of sessions
+					Size: ^[batch_size]^
+				* A_profiles (list): A 2-d list, the profiles (a 1-d list with size 4) of speaker ^A^.
+					Size: ^[batch_size, 4]^
+				* B_profiles (list): A 2-d list, the profiles (a 1-d list with size 4) of speaker ^B^.
+					Size: ^[batch_size, 4]^
+				* floors (list): A 2-d list, speakers of utterances. The inner lists may have different length.
+					Size: ^[batch_size, _utt_len]^
+					Examples: [['A', 'B', 'A'],
+							   ['B', 'A']]
+				* dialog_acts (list): A 2-d list, dialog act ids of utterances.
+									The inner lists may have different length.
+					Size: ^[batch_size, _utt_len]^
+
+			See the example belows.
+
+		Examples:
+			>>> dataloader.get_batch('train', [1])
+			>>>
+
+		Todo:
+			* fix the missing example
+		'''
+		res = {}
+		res['session'] = super(SwitchboardCorpus, self).get_batch(key, index)
+		gather = lambda sub_key: [self.data[key][sub_key][i] for i in index]
+		for sub_key in self.data[key]:
+			if sub_key != 'session':
+				res[sub_key] = gather(sub_key)
+		return res
+
+	def _cut_batch_data(self, batch_data, start, end):
+		'''Using session[start: end - 1) as context, session[end - 1] as response,
+			invoked by ^SwitchboardCorpus.split_session^
+		'''
+		raw_batch = {}
+		raw_batch['topics'] = batch_data['topics']
+		raw_batch['A_profiles'] = batch_data['A_profiles']
+		raw_batch['B_profiles'] = batch_data['B_profiles']
+		keys = ['floors', 'dialog_acts', 'speakers', 'posts_length', 'responses_length']
+		for key in keys:
+			raw_batch[key] = []
+		for i in range(len(batch_data['topics'])):
+			raw_batch['floors'].append(batch_data['floors'][i][start: end - 1])
+			raw_batch['posts_length'].append( \
+				batch_data['session']['sent_length'][i][start: end - 1])
+			utt_len = len(batch_data['floors'][i])
+			raw_batch['dialog_acts'].append( \
+				batch_data['dialog_acts'][i][min(utt_len - 1, end - 1)])
+			raw_batch['speakers'].append( \
+				batch_data['floors'][i][min(utt_len - 1, end - 1)])
+			if end - 1 < utt_len:
+				raw_batch['responses_length'].append( \
+					batch_data['session']['sent_length'][i][end - 1])
+			else:
+				raw_batch['responses_length'].append(1)
+		raw_batch['contexts'] = batch_data['session']['sent'][:, start: end - 1]
+		raw_batch['responses'] = batch_data['session']['sent'][:, end - 1]
+		return raw_batch
+
+	def split_session(self, batch_data, inference=False):
```

Comment:
Move it to your model, if you just want to use it. I will consider to make a api like this, but not now.